google/cloud/bigquery/table.py

```diff
@@ -1011,6 +1011,13 @@ def _build_resource(self, filter_fields):
         """Generate a resource for ``update``."""
         return _helpers._build_resource_from_properties(self, filter_fields)
 
+    def __eq__(self, other):
+        if not isinstance(other, Table):
+            return NotImplemented
+        return self._properties == other._properties
```

Comment:
I think this is overly stringent.

```
>>> from google.cloud.bigquery import table
>>> t1 = table.Table("precise-truck-742.riversnake.lake")
>>> t2 = client.get_table("precise-truck-742.riversnake.lake")
>>> t1.__class__, t2.__class__
(<class 'google.cloud.bigquery.table.Table'>, <class 'google.cloud.bigquery.table.Table'>)
>>> t1._properties == t2._properties
False
```

`t1` and `t2` refer to the same table, and should be equal IMO.  They only differ by meta data that is dynamic. `t2` has extra data that is effectively cached.

I think you should just compare the table reference.

Reply:
I was also considering that option, yes. And now you even made a convincing case with `t2` fetched by the client, I also think it should be equal to `t1`.